packages/alert-banner/index.tsx

```diff
@@ -71,32 +80,54 @@ function AlertBanner({
   }
 
   return (
-    <div
-      className={classNames(
-        s.root,
-        themeClass,
-        { [s.isShown]: isShown },
-        { [s.hideOnMobile]: hideOnMobile }
-      )}
-    >
-      <a
-        href={url}
-        className={s.linkElem}
-        onClick={() => analytics.trackClick({ linkText, product, tag, text })}
+    <>
+      <div
+        className={classNames(
+          s.root,
+          themeClass,
+          { [s.isShown]: isShown },
+          { [s.hideOnMobile]: hideOnMobile }
+        )}
+        suppressHydrationWarning
       >
-        <span className={s.textContainer}>
-          <span className={s.tag}>{tag} </span>
-          <span className={s.text}>
-            {text}
-            {linkText ? <span className={s.linkText}> {linkText}</span> : null}
+        <a
+          href={url}
+          className={s.linkElem}
+          onClick={() => analytics.trackClick({ linkText, product, tag, text })}
+        >
+          <span className={s.textContainer}>
+            <span className={s.tag}>{tag} </span>
+            <span className={s.text}>
+              {text}
+              {linkText ? (
+                <span className={s.linkText}> {linkText}</span>
+              ) : null}
+            </span>
           </span>
-        </span>
-      </a>
-      <button className={s.closeButton} onClick={closeBanner}>
-        <InlineSvg src={CloseIcon} />
-        <VisuallyHidden>Dismiss alert</VisuallyHidden>
-      </button>
-    </div>
+        </a>
+        <button className={s.closeButton} onClick={closeBanner}>
+          <InlineSvg src={CloseIcon} />
+          <VisuallyHidden>Dismiss alert</VisuallyHidden>
+        </button>
+      </div>
+      <script
```

Comment:
@BRKalow In what instance is this inline script run? Is this to handle when the server-side HTML contains the alert banner but it's been dismissed by the user? If that's the case, shouldn't we switch the logic such that the server always returns the alert hidden, and it then is shown once the `useState` hook value is calculated on the client?

Reply:
@dstaley would that cause another layout shift though when we try to render the alert?

Reply:
@dstaley yes that scenario is correct. wouldn't we have layout shift still? With this check, the isShown class is removed before the browser paints and so it removes the shift. If we wait until the state value is initialized there would be a shift I think.

Reply:
Ahhh, okay I see what's happening here! I'm not thrilled at the idea of a render-blocking inline script, but I agree that's preferred to layout shift.

Reply:
@dstaley Yeah in general I agree, but in this scenario it seems like a reasonable tradeoff 🤞